### PR TITLE
[PHPBB3/13903] replacing regexp path naming for container by md5

### PR DESCRIPTION
**Pull request by Wardormeur**
[PHPBB3/13903] Replacement of the path creation based on a blacklist by a md5 filename to support further configurations

PHPBB3-13903

### CODE REVIEW

File: phpBB/phpbb/di/container_builder.php [hunk 1]
```diff
@@ -399,7 +399,6 @@ protected function inject_custom_parameters()
 	*/
 	protected function get_container_filename()
 	{
-		$filename = str_replace(array('/', '.'), array('slash', 'dot'), $this->phpbb_root_path);
-		return $this->phpbb_root_path . 'cache/container_' . $filename . '.' . $this->php_ext;
+		return $this->phpbb_root_path . 'cache/container_' . md5($filename) . '.' . $this->php_ext;
```
Comment (marc1706) [inline phpBB/phpbb/di/container_builder.php]: $filename is undefined now

**Comment by Nicofuma** [comment]
The commit message is still wrong ;)

It should be:
```
[ticket/13903] Replacing regexp path naming for container by md5
Replacement of the path creation based on a blacklist by a md5 filename to support further configurations

PHPBB3-13903
```

**Comment by Nicofuma** [comment]
otherwise +1

**Comment by Wardormeur** [comment]
I'm amazed to see he was still able to create a container without the filename (md5 of null?). Fixing it atm, sorry about that.

**Comment by Nicofuma** [comment]
;)

**Comment by nickvergessen** [comment]
Why is this better then the version from before?
The previous one allowed you to quickly identify which container is for which path?

**Comment by Wardormeur** [comment]
Because it disallows container creation with windows full path, as the ':' isn't parsed. So either you thinks of looking for every "edge" case and parse it, either you go for a more robust and all-around solution

**Comment by bantu** [comment]
I very much prefer this version over the old version.

**Comment by bantu** [comment]
:+1:

**Comment by bantu** [comment]
Can you please squash the two commits together and make the commit message:
```
 [ticket/13903] Replacing regexp path naming for container by md5

Replacement of the path creation based on a blacklist by a md5 filename to
support broader configurations

PHPBB3-13903
```

**Comment by bantu** [comment]
Replaced by #3711

**Comment by Wardormeur** [comment]
Sorry about that, haven't been on my projects lately for IRL reasons. Thanks for supporting this change :)

**Pull request closed without merging**